index.js

```diff
@@ -0,0 +1,142 @@
+axios.defaults.baseURL = 'https://trektravel.herokuapp.com/trips';
+//
+// Status Management
+//
+const reportStatus = (message) => {
+  $('#status-message').html(message);
+  $('#status-message').removeClass('alert alert-danger')
+  $('#status-message').addClass('alert alert-success')
+};
+const errorStyling = () => {
+  $('#status-message').removeClass('alert alert-success')
+  $('#status-message').addClass('alert alert-danger')
+}
+
+
+const reportError = (message, errors) => {
+  let content = `<p>${message}</p><ul>`;
+  for (const field in errors) {
+    for (const problem of errors[field]) {
+      content += `<li>${field}: ${problem}</li>`;
+    }
+  }
+  content += "</ul>";
+  reportStatus(content);
+};
+
+
+//
+// Loading Trips
+//
+const loadTrips = () => {
+  reportStatus('Loading trips...');
+
+  const tripList = $('#trip-list');
+  tripList.empty();
+  $('h2').removeClass('hidden')
+
+  axios.get()
+    .then((response) => {
+      reportStatus(`Successfully loaded ${response.data.length} trips`);
+      response.data.forEach((trip) => {
+        const listItem = $(`<li>${trip.name}</li>`);
+        tripList.append(listItem);
+        listItem.click(showTripDetails(trip.id));
+      });
+    })
+    .catch((error) => {
+      reportStatus(`Encountered an error while loading trips: ${error.message}`);
+      errorStyling();
+      console.log(error);
+    });
+};
+
+//Load Trip Details
+const showTripDetails = (tripId) => {
+  const loadTripDetails = () => {
+    reportStatus('Loading trip details...');
+    $('.details').removeClass('hidden')
```

Comment:
Good use of a closure here to track the trip ID.

As a style note, you would typically write the outer function as an anonymous function where it's needed (the success callback for `loadTrips`).